data/campaigns/atl02.wmf/scripting/texts.lua

```diff
@@ -0,0 +1,763 @@
+         .. new_objectives(obj_explore, obj_expand),
+      h=500,
+   },
+} -- end of initial messages.
+
+spidercloth_1 = {
+   {
+      title = _("Opol visits Jundlina"),
+      body = opol(
+         -- TRANSLATORS: Opol
+         _([[May Satul warm you, Jundlina. As the highest weaver of the guild I have bad news for you. I just realized that we are running out of spidercloth. This is a most perilous case, for as you know, spidercloth is needed for many buildings.]])
+         .. paragraphdivider() ..
+         -- TRANSLATORS: Opol
+         _([[We should build up a spidercloth production urgently.]]))
+   },
+   {
+       title = _("Jundlina Replies"),
+       body = jundlina(_("Jundlina"),
+         -- TRANSLATORS: Jundlina
+         _([[May Satul warm you too, Opol. I wanted to delay production of spidercloth, but I understand the urgency. We need to build a complete spidercloth production instantly.]]))
+         .. new_objectives(obj_spidercloth)
+   }
+}
+
+spidercloth_2 = {
+   {
+      title = _("Opol makes a point"),
+      body = opol(
+         -- TRANSLATORS: Opol
+         _([[Jundlina! It seems there is another problem even more urgent. I fear we haven’t built a farm to feed our spiders yet. As this is one of the buildings needing spidercloth for completion, not completing one in time will result in a crisis that can never be resolved.]]))
+   },
+   {
+      title = _("Jundlina Acknowledges"),
+      body = jundlina(_("Jundlina"),
+         -- TRANSLATORS: Jundlina
+         _([[I see the risk, Opol. Thank you for reporting. We will build a farm immediately.]]))
+   }
+}
+
+spidercloth_3 = {
+   {
+      title = _("Opol Seeks Out Jundlina"),
+      body = opol(
+         -- TRANSLATORS: Opol
+         _([[Jundlina, may Satul warm you! The weaving mill and spider farm are complete and the weavers’ guild can start their work again. I promise we will deliver the finest cloth and some tabards soon. Maybe we can build a gold spinning mill later on to make better tabards as well.]]))
+   },
+   {
+      title = _("Jundlina Cheers"),
+      body = jundlina(_("Jundlina"),
+         -- TRANSLATORS: Jundlina
+         _([[This is good news indeed, Opol. You may go back to your work now, and may Satul warm you and the weavers’ guild!]]))
+   }
+}
+
+spidercloth_3a = {
+   {
+      title = _("Opol Seeks Out Sidolus"),
+      body = opol(
+         -- TRANSLATORS: Opol
+         _([[Sidolus, may Satul warm you! The weaving mill and spider farm are complete and the weavers’ guild can start their work again. I promise we will deliver the finest cloth and some tabards soon. Maybe we can build a gold spinning mill later on to make better tabards as well.]]))
+   },
+   {
+      title = _("Sidolus Cheers"),
+      body = sidolus(
+         -- TRANSLATORS: Sidolus
+         _([[This is good news indeed, Opol. You may go back to your work now, and may Satul warm you and the weavers’ guild!]]))
+   }
+}
```

Comment:
Couldn't these be simplified by using a `leader` variable in `mission_thread.lua`? I guess some languages will need different versions of the titles and the greetings, but those can be indexed by `leader`. So it would become something like:

```
-- these functions need to be harmonised, because jundlina() takes one more argument 
speak.Jundlina = jundlina
speak.Sidolus = sidolus

greeting.Jundlina = _("Jundlina, may Satul warm you!")
greeting.Sidolus = _("Sidolus, may Satul warm you!")

cheer.Jundlina = _("Jundlina Cheers")
cheer.Sidolus = _("Sidolus Cheers")

…

function spidercloth3(leader)
   spider3_title.Jundlina = _("Opol Seeks out Jundlina")
   spider3_title.Sidolus = _("Opol Seeks out Sidolus")
   return {
      {
         title = spider3_title[leader],
         body = opol(greeting[leader] .. _([[The weaving mill and spider farm etc. etc. better tabards as well.]]))
      },
      {
         title = cheer[leader],
         body = speak[leader](_([[This is good news indeed, Opol. etc. etc. you and the weavers’ guild!]]))
      }
   }
end
```

So the long texts wouldn't be duplicated and `mission_thread.lua` would need fewer checks whether a given objective is fulfilled before Jundlina is taken hostage.